packages/story-editor/src/components/toggleButton/index.js

```diff
@@ -35,57 +39,135 @@ import { forwardRef } from '@googleforcreators/react';
  */
 import Tooltip from '../tooltip';
 
+const buttonSize = 36;
+
 const Button = styled(dsButton)`
-  height: 36px;
-  width: ${({ isSquare }) => (isSquare ? 36 : 58)}px;
-  display: inline-block;
-  border: 1px solid ${({ theme }) => theme.colors.border.defaultNormal};
-  padding: 2px 0;
-  color: ${({ theme }) => theme.colors.fg.primary};
-  ${({ isOpen, theme }) =>
-    isOpen &&
-    css`
-      border-color: ${theme.colors.bg.secondary};
-      background-color: ${theme.colors.bg.secondary};
-    `}
-  ${({ hasText, theme }) =>
-    hasText &&
-    css`
-      padding: 2px 0 2px 8px;
-      width: auto;
-      ${themeHelpers.expandPresetStyles({
-        preset: {
-          ...theme.typography.presets.paragraph[
-            THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL
-          ],
-        },
-        theme,
-      })};
-
-      span {
-        padding-left: 6px;
-      }
-    `}
+  width: auto;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  border-width: 1px;
+  border-style: solid;
+  padding: 0 10px;
+  white-space: nowrap;
+  // Handle props-dependent styling in one place
+  ${({ isOpen, hasText, $size = buttonSize, theme }) => css`
+    height: ${resolveSizeUnits($size)};
+    min-width: ${resolveSizeUnits($size)};
+    color: ${theme.colors.fg.primary};
+    border-color: ${
+      isOpen ? theme.colors.bg.secondary : theme.colors.border.defaultNormal
+    };
+    background-color: ${
+      isOpen ? theme.colors.bg.secondary : theme.colors.bg.primary
+    };
+    ${
+      hasText &&
+      css`
+        ${themeHelpers.expandPresetStyles({
+          preset: {
+            ...theme.typography.presets.paragraph[
+              THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL
+            ],
+          },
+          theme,
+        })};
+
+        .badge {
+          margin: 0 8px;
+        }
+      `
+    }
+
+    .badge.open,
+    &:hover .badge,
+    &:focus .badge {
```

Comment:
Again, there's a lot of classes added here. Is this the best way to do this?

Reply:
Old habits… It can be cleaner, but deviates from patterns used elsewhere.